src/NServiceBus.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -174,74 +174,53 @@ void Connection_ConnectionShutdown(object sender, ShutdownEventArgs e)
 
                 if (rejectMessage)
                 {
-                    await RejectMessage(channel, message.DeliveryTag, messageId).ConfigureAwait(false);
+                    await RejectMessage(channel, message.DeliveryTag).ConfigureAwait(false);
                 }
                 else
                 {
-                    await AcknowledgeMessage(channel, message.DeliveryTag, messageId).ConfigureAwait(false);
+                    await AcknowledgeMessage(channel, message.DeliveryTag).ConfigureAwait(false);
                 }
             }
+            catch (AlreadyClosedException ex)
+            {
+                Logger.Warn($"Attempt to acknowledge/reject message {messageId} failed because the channel was closed. The message will be requeued.", ex);
+            }
             catch (Exception ex)
             {
-                Logger.Warn($"Error while attempting to process message {messageId}. The message will be rejected.", ex);
+                Logger.Warn($"Error while attempting to process message {messageId}. The message will be requeued.", ex);
 
-                await RejectMessage(channel, message.DeliveryTag, messageId).ConfigureAwait(false);
+                try
+                {
+                    await RejectMessage(channel, message.DeliveryTag).ConfigureAwait(false);
+                }
+                catch (AlreadyClosedException) { }
```

Comment:
After reading [your response to my initial question](https://github.com/Particular/NServiceBus.RabbitMQ/pull/165#discussion_r68841837) about this I now understand why it's OK to swallow this exception here.

However, in the old version of the code, we would have executed this in this case:

```C#
Logger.Warn($"Attempt to reject message {messageId} failed because the channel was closed. The message will be requeued.", ex);
```

Is it worth preserving that? We could also preserve the "rejected" instead of the "requeued" in the log message above.